### PR TITLE
[BCNM] Add Waking the Beast Ifrit BCNM

### PR DESCRIPTION
**Pull request by TracentEden2**
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my PR will be ignored.
- [x] I understand I should leave resolving conversations to the LandSandBoat team so that reviewers won't miss what was said.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have _**tested my code and the things my code has changed**_ since the last commit in the PR and will test after any later commits.

## What does this pull request do?
This PR adds the waking the beast Ifrit BCNM fight. This is the first in a series of PRs for adding WTB. Note that this does not add the actual quest so players cannot yet enter the BCNM fight, as my plan is to add all of the BCNM fights before a final PR that adds in the quest. Also note that the commented out `-- mob:setMobMod(xi.mobMod.SKIP_ALLEGIANCE_CHECK, 1)` will be enabled once this [related PR](https://github.com/LandSandBoat/server/pull/6124) that adds the specific mob mod is merged.

I am not sure if this should be marked as experimental or not?

The info for the PR has been taken from a variety of captures and retail videos including:
- General behavior, mobskills, HP, immunities, res ranks, and MDT ([here](https://www.youtube.com/watch?v=i9dTJ-A0MCg), [here](https://youtu.be/poednY37W1Q) see [capture logs](https://www.dropbox.com/sh/cge8c766pmam3ha/AAAb2ROna7OPu-e6eCPJtLcIa?dl=0) as chat log cutoff)
- Elementals refocusing hate on avatar target periodically ([here](https://youtu.be/A2SKt45n578?t=27))
- Elemental En-Plague Power ([here](https://www.youtube.com/watch?v=8EbAhDq5SFA&t=11s))

## Steps to test these changes
As a GM go to the BCNM and enter the fight and fight the mobs

### CODE REVIEW

File: scripts/zones/Cloister_of_Flames/mobs/Ifrit_Prime_WTB.lua
Comment (Frankie-hz) [inline scripts/zones/Cloister_of_Flames/mobs/Ifrit_Prime_WTB.lua:82]: I am a big fan of ternary expressions, but this is probably a personal preference because I hate if statements ha. Curious what others think though. More for my curiosity than anything (code is perfectly fine).

`spellTarget = math.random(1, 2) == 1 and mob or elemental`

File: scripts/battlefields/Cloister_of_Flames/waking_the_beast.lua
Comment (claywar) [inline scripts/battlefields/Cloister_of_Flames/waking_the_beast.lua:23]: Not sure if Waking the Beast was converted to interaction yet, but if so, this could most likely move there

File: scripts/battlefields/Cloister_of_Flames/waking_the_beast.lua
Comment (claywar) [inline scripts/battlefields/Cloister_of_Flames/waking_the_beast.lua:53]: Do these start spawned? If not, then spawned will need to be set to false for this group

File: scripts/zones/Cloister_of_Flames/mobs/Ifrit_Prime_WTB.lua
Comment (claywar) [inline scripts/zones/Cloister_of_Flames/mobs/Ifrit_Prime_WTB.lua:115]: This cleanup can probably move to the battlefield script where you're setting the battlefield to win status

**Review by claywar — COMMENTED** [review]
Looks reasonable to me.

File: scripts/battlefields/Cloister_of_Flames/waking_the_beast.lua
Comment (TracentEden2) [reply scripts/battlefields/Cloister_of_Flames/waking_the_beast.lua:23]: > Not sure if Waking the Beast was converted to interaction yet, but if so, this could most likely move there

Ah, yeah I think the quest is not implemented yet. I have IF code for the quest but was going to wait until all BCNMs are merged to make the PR with the quest itself.

File: scripts/battlefields/Cloister_of_Flames/waking_the_beast.lua
Comment (TracentEden2) [reply scripts/battlefields/Cloister_of_Flames/waking_the_beast.lua:53]: > Do these start spawned? If not, then spawned will need to be set to false for this group

Yes, they start out spawned, and if killed then can be respawned by Ifrit.

File: scripts/zones/Cloister_of_Flames/mobs/Ifrit_Prime_WTB.lua
Comment (TracentEden2) [reply scripts/zones/Cloister_of_Flames/mobs/Ifrit_Prime_WTB.lua:115]: > This cleanup can probably move to the battlefield script where you're setting the battlefield to win status

Yep, I moved this now to the battlefield script, thanks!

File: scripts/zones/Cloister_of_Flames/mobs/Ifrit_Prime_WTB.lua
Comment (TracentEden2) [reply scripts/zones/Cloister_of_Flames/mobs/Ifrit_Prime_WTB.lua:82]: > I am a big fan of ternary expressions, but this is probably a personal preference because I hate if statements ha. Curious what others think though. More for my curiosity than anything (code is perfectly fine).
> 
> `spellTarget = math.random(1, 2) == 1 and mob or elemental`

Yeah, I like the if statements mainly for readability, especially for new devs or contributors since this is in Lua.

**Review by claywar — APPROVED** [review]

**Review by Xaver-DaRed — APPROVED** [review]

**Pull request merged**
